gc.c

```diff
@@ -4841,8 +4840,8 @@ obj_memsize_of(VALUE obj, int use_all_types)
         if (use_all_types) size += rb_objspace_data_type_memsize(obj);
         break;
       case T_MATCH:
-        if (RMATCH(obj)->rmatch) {
-            struct rmatch *rm = RMATCH(obj)->rmatch;
+        {
+            struct rmatch *rm = &RMATCH_EXT(obj)->rmatch;
             size += onig_region_memsize(&rm->regs);
             size += sizeof(struct rmatch_offset) * rm->char_offset_num_allocated;
             size += sizeof(struct rmatch);
```

Comment:
We no longer need to add the `sizeof(struct rmatch)` because that size is part of the slot.